repos/fdbt-site/src/pages/api/updateFareTriangle.ts

```diff
@@ -0,0 +1,158 @@
+};
+
+const checkFareStageNames = (fareTriangleData: UserFareStages, fareZoneNames: string[]): boolean => {
+    const thereIsANameMismatch = fareTriangleData.fareStages.some((fs) => {
+        return !fareZoneNames.includes(fs.stageName);
+    });
+
+    return thereIsANameMismatch;
+};
+
+const getStageCountMismatchError = () => {
+    const errors: ErrorInfo[] = [
+        {
+            id: errorId,
+            errorMessage:
+                'The number of fare stages of your updated fares triangle do not match the one you have previously uploaded. Update your triangle, ensuring the number of fare stages match before trying to upload again',
+        },
+    ];
+
+    return errors;
+};
+
+const getNameMismatchError = () => {
+    const errors: ErrorInfo[] = [
+        {
+            id: errorId,
+            errorMessage:
+                'The name of one or more fare stages of your updated fares triangle does not match what you had have previously uploaded. Update your triangle, ensuring the names of fare stages match before trying to upload again',
+        },
+    ];
+
+    return errors;
+};
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        const ticket = getSessionAttribute(req, MATCHING_JSON_ATTRIBUTE) as
+            | WithIds<SingleTicket>
+            | WithIds<ReturnTicket>;
+
+        const productAndServiceId = getSessionAttribute(req, PRODUCT_AND_SERVICE_ID_ATTRIBUTE);
+
+        if (ticket === undefined) {
+            throw new Error('Could not find the matching json in the session');
+        }
+
+        const formData = await getFormData(req);
+        const { fields } = formData;
+
+        if (!fields?.poundsOrPence) {
+            const errors: ErrorInfo[] = [
+                { id: 'pounds', errorMessage: 'You must select whether the prices are in pounds or pence' },
+            ];
+
+            setCsvUploadAttributeAndRedirect(req, res, errors);
+
+            return;
+        }
+
+        const { poundsOrPence } = fields;
+
+        const { fileContents, fileError } = await processFileUpload(formData, 'csv-upload');
+
+        if (fileError) {
+            const errors: ErrorInfo[] = [{ id: errorId, errorMessage: fileError }];
+
+            setCsvUploadAttributeAndRedirect(req, res, errors, fields.poundsOrPence as string);
+
+            return;
+        }
+
+        if (fileContents) {
+            const uuid = ticket.uuid;
+
+            const fareZoneNames = getNamesOfFareZones(ticket);
+
+            const fareTriangleData = faresTriangleDataMapper(fileContents, req, res, poundsOrPence as string);
+
+            if (!fareTriangleData || !fareZoneNames) {
+                return;
+            }
+
+            let errors;
+
+            // check to see if the the number of fare stages does not match
+            if (fareTriangleData.fareStages.length !== fareZoneNames.length) {
+                errors = getStageCountMismatchError();
```

Comment:
The way this is written, if the errors array gets set here due to there being a different number of fare stages, it will get overwritten on line 131. I think we might need to check to see if errors.length > 0 and if it is, redirect before we do the 2nd error check